packages/training/python/tests/test_enhanced_rewards_extended.py

```
@@ -0,0 +1,1130 @@
+
+class TestIsTradingActionComprehensive:
+    """Comprehensive tests for is_trading_action."""
+    
+    def test_all_known_trading_actions(self):
+        """All explicitly listed trading actions."""
+        for action in TRADING_ACTION_TYPES:
+            assert is_trading_action(action), f"{action} should be trading"
+    
+    def test_case_variations(self):
+        """Case insensitivity."""
+        assert is_trading_action("BUY") == True
```

Ruff (0.14.11)

Avoid equality comparisons to `True`; use `is_trading_action("BUY"):` for truth checks

Replace with `is_trading_action("BUY")`

(E712)